### PR TITLE
feat(curriculum): add the JavaScript Array Quiz

### LINKED ISSUE #56170: Welcome to Hacktoberfest 2024
# What is Hacktoberfest?

[Hacktoberfest](https://hacktoberfest.com) is a month-long celebration of open source. Projects, such as freeCodeCamp, can opt-in to participate. When they have done so, contributors (just like you!) can make pull requests to the project that are eligible for the event.

Contributors who have made four eligible pull requests (PRs which receive an "approve" review or are merged into the default branch) can earn a digital badge through Holopin - this is customisable and evolves over time.

> [!NOTE]
> Contributors who have a PR merged for this issue may also reach out to Naomi on Discord to claim a special Hacktoberfest 2024 role.

# How to Participate

We are in the process of overhauling our interactive curriculum. This change will include new lesson formats such as videos, lectures, interactive labs and workshops, and more.

Part of this initiative requires creating knowledge comprehension quizzes. We are asking contributors to participate in Hacktoberfest by making pull requests to create the content for these quizzes.

We need pull requests for each of the following quizzes.

<details>
<summary><h2>Show List</h2></summary>

```[tasklist]
### Tasks
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56174
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56175
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56176
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56177
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56178
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56179
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56180
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56181
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56182
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56183
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56184
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56185
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56186
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56187
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56188
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56189
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56190
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56191
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56192
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56193
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56194
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56195
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56196
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56197
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56198
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56199
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56200
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56201
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56202
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56203
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56204
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56205
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56206
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56207
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56208
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56209
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56210
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56211
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56212
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56213
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56214
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56215
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56216
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56217
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56218
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56219
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56220
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56221
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56222
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56223
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56224
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56225
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56226
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56227
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56228
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56229
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56230
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56231
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56232
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56233
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56234
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56235
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56236
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56237
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/issues/56238
```

</details>

> [!IMPORTANT]
> To allow as many contributors to participate as possible, we kindly ask that:
> - Each pull request create exactly one (1) quiz
> - Each contributor create no more than one pull request for this issue

# How to Write a Quiz

All quizzes can be found in the new [`front-end-development`](https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/25-front-end-development) directory.

A quiz is contained in a single markdown file, such as [the Design Fundamentals quiz](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/quiz-design-fundamentals/66ed901ff45ce3ece4053eba.md).

These markdown files use a template:

````md
---
id: id-goes-here
title: Semantic HTML Quiz
challengeType: 8
dashedName: semantic-html-quiz
---

# --description--

Answer all of the questions below correctly to pass the quiz.

# --quizzes--

## --quiz--

### --question--

#### --text--

Placeholder question

#### --distractors--

Placeholder distractor 1

---

Placeholder distractor 2

---

Placeholder distractor 3

#### --answer--

Placeholder answer
````

Quizzes are all multiple choice. The "distractors" are incorrect answers to the question. You may list the distractors in any order. The `--answer--` should be the answer which is correct.

A quiz should have a total of 20 questions related to the topic. We will attempt to provide a list of sub-topics to choose from before the event starts.

# Examples

Here is an example of what a completed quiz file should look like: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/quiz-basic-html/66df3b712c41c499e9d31e5b.md

Here is an example of a pull request which follows our contributing guidelines might look like: https://github.com/freeCodeCamp/freeCodeCamp/pull/56417

# Plagiarism and AI-generated Content

Do **NOT** copy questions directly from other sources, even if paraphrased. Questions you submit should be your own creations. This includes using generative AI to create questions - please do NOT use AI (ChatGPT, Claude, etc) to create questions.

# Interested in Contributing?

> [!TIP]
> Before contributing, be sure to read our [documentation](https://contribute.freecodecamp.org) thoroughly. If you have any questions, reach out to us in [Discord](https://discord.gg/KVUmVXA) and we will be more than happy to assist you.

If you are interested in contributing, please respond OH THE ISSUE YOU'D LIKE TO WORK ON. We will then add you to the issue so other contributors know you have claimed that quiz.

> [!CAUTION]
> If you have claimed a quiz, and we have seen no activity within 7 days, we will release the quiz so another contributor may claim it.

We look forward to seeing your contributions. Happy Coding! 💜

### LINKED ISSUE #56201: [HACKTOBERFEST 2024] - Arrays Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Arrays quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-arrays`.

**All quiz questions need to relate to one of these topics:**

- key characteristics of arrays
- difference between primitive and non-primitive data types
- accessing and updating elements
- push() and pop()
- .length
- 1D and 2D arrays
- destructuring (rest syntax)
- indexOf()
- splice()
- includes()
- concat()
- splice()
- spread operator
- reverse a string

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by Ritesh2351235**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56201

### CODE REVIEW

**Comment by jdwilkin4** [comment]
You are linking to the wrong issue.

Make sure to update your PR description to link to the issue you were assigned to

**Comment by ilenia-magoni** [comment]
the correct issue to link is https://github.com/freeCodeCamp/freeCodeCamp/issues/56201

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 1]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder question
+What will be the result of the following array method `indexOf()`?
+
+
+```javascript
+const numbers = [10, 20, 30, 40];
+const index = numbers.indexOf(20);
+console.log(index);
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:71]:
````suggestion
What will be the result of the following array method `indexOf()`?

```js
const numbers = [10, 20, 30, 40];
const index = numbers.indexOf(20);
console.log(index);
```
````

I think 'javascript' is also accepted by the linter, but we usually use 'js'

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 2]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder question
+What will the following code output using the spread operator?
+
+
+```javascript
+const arr1 = [1, 2, 3];
+const arr2 = [...arr1, 4, 5];
+console.log(arr2);
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:100]:
````suggestion
What will the following code output using the spread operator?

```js
const arr1 = [1, 2, 3];
const arr2 = [...arr1, 4, 5];
console.log(arr2);
```
````

**Comment by Ritesh2351235** [comment]
@jdwilkin4 Sorry for the mistake

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 3]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+What will `splice()` do in the following example?
+
+
+```javascript
+
+const colors = ['red', 'blue', 'green', 'yellow'];
+colors.splice(1, 2, 'purple');
+console.log(colors);
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:131]:
````suggestion
What will `splice()` do in the following example?

```js
const colors = ['red', 'blue', 'green', 'yellow'];
colors.splice(1, 2, 'purple');
console.log(colors);
```
````

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 4]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+What will be the result of the following code using `slice()`?
+
+
+```javascript
+
+const arr = ['apple', 'banana', 'cherry', 'date'];
+const slicedArr = arr.slice(1, 3);
+console.log(slicedArr);
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:162]:
````suggestion
What will be the result of the following code using `slice()`?

```javascript
const arr = ['apple', 'banana', 'cherry', 'date'];
const slicedArr = arr.slice(1, 3);
console.log(slicedArr);
```
````

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 5]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+```javascript
+
+const fruits = ['apple', 'banana', 'cherry', 'apple', 'orange'];
+const index = fruits.indexOf('apple');
+if (index !== -1) {
+    fruits.splice(index, 1);
+}
+console.log(fruits);
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:285]:
````suggestion
What will be the output of this code ?

```js
const fruits = ['apple', 'banana', 'cherry', 'apple', 'orange'];
const index = fruits.indexOf('apple');
if (index !== -1) {
    fruits.splice(index, 1);
}
console.log(fruits);
```
````

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 6]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+
+```javascript
+
+const arr1 = [1, 2];
+const arr2 = [3, 4, 5];
+const combined = arr1.concat(arr2);
+const result = combined.includes(3);
+console.log(result);
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:318]:
````suggestion
What is the final value of result in the following code ?

```js
const arr1 = [1, 2];
const arr2 = [3, 4, 5];
const combined = arr1.concat(arr2);
const result = combined.includes(3);
console.log(result);
```
````

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 7]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+What does the following code output?
+
+
+```javascript
+
+const nums = [1, 2, 3, 4];
+const doubled = nums.map(n => n * 2);
+console.log(doubled.includes(8));
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:349]:
````suggestion
What does the following code output?

```ja
const nums = [1, 2, 3, 4];
const doubled = nums.map(n => n * 2);
console.log(doubled.includes(8));
```
````

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 8]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+
+let matrix = [
+  [1, 2, 3],
+  [4, 5, 6],
+  [7, 8, 9]
+];
+matrix.splice(1, 1);
+console.log(matrix);
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:384]:
````suggestion
What is the output when using splice() to remove elements from a 2D array?

```js
let matrix = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9]
];
matrix.splice(1, 1);
console.log(matrix);
```
````

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 9]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --text--
 
-Placeholder question
+How does the rest parameter (`...rest`) work in array destructuring in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+It is used to add extra elements to the array.
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:477]:
```suggestion
It is used to add extra elements to the array.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 10]
```diff
@@ -17,439 +17,529 @@ Answer all of the questions below correctly to pass the quiz.
...
+```javascript
+
+const numbers = [10, 20, 30];
+numbers.push(40);
+numbers.pop();
+const length = numbers.length;
+console.log(length);
+
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:506]:
````suggestion
What will be the result of this code?

```js
const numbers = [10, 20, 30];
numbers.push(40);
numbers.pop();
const length = numbers.length;
console.log(length);
```
````

**Review by majestic-owl448 — COMMENTED** [review]
mostly empty lines suggestions

**Comment by Ritesh2351235** [comment]
@ilenia-magoni I have made the necessary changes can you please check it out once.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 11]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --text--
 
-Placeholder question
+What will be the result of the following array method `indexOf()`?
+
+```js
+const numbers = [10, 20, 30, 40];
+const index = numbers.indexOf(20);
+console.log(index);
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:70]:
```suggestion
What value will be assigned to the `index` variable?

```js
const numbers = [10, 20, 30, 40];
const index = numbers.indexOf(20);
```
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 12]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --text--
 
-Placeholder question
+What will the following code output using the spread operator?
+
+```js
+const arr1 = [1, 2, 3];
+const arr2 = [...arr1, 4, 5];
+console.log(arr2);
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:98]:
```suggestion
What value will be assigned to the `arr2` variable?

```js
const arr1 = [1, 2, 3];
const arr2 = [...arr1, 4, 5];
```
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 13]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --text--
 
-Placeholder question
+What will `splice()` do in the following example?
+
+```js
+const colors = ['red', 'blue', 'green', 'yellow'];
+colors.splice(1, 2, 'purple');
+console.log(colors);
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:126]:
```suggestion
What will this code print to the console?

```js
const colors = ['red', 'blue', 'green', 'yellow'];
colors.splice(1, 2, 'purple');
console.log(colors);
```
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 14]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --text--
 
-Placeholder question
+What will be the result of the following code using `slice()`?
+
+```js
+const arr = ['apple', 'banana', 'cherry', 'date'];
+const slicedArr = arr.slice(1, 3);
+console.log(slicedArr);
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:154]:
```suggestion
What value will be assigned to the `slicedArr` variable?

```js
const arr = ['apple', 'banana', 'cherry', 'date'];
const slicedArr = arr.slice(1, 3);
```
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 15]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+Nothing happens; the array remains the same size.
 
 #### --answer--
 
-Placeholder answer
+The array gets truncated, removing elements beyond the new length.
+
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:259]:
```suggestion
The array gets truncated, removing elements beyond the new length.
```

Extra new line.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 16]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`['banana', 'cherry', 'apple', 'orange']`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the final value of result in the following code ?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:296]:
```suggestion
What is the final value of `result` in the following code ?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 17]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
+```js
+const nums = [1, 2, 3, 4];
+const doubled = nums.map(n => n * 2);
+console.log(doubled.includes(8));
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+false
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:336]:
```suggestion
`false`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 18]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+`[2, 4, 6, 8]`
 
 ---
 
-Placeholder distractor 3
+Error
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:344]:
```suggestion
`Error`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 19]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
+const arr2 = [3, 4, 5];
+const combined = arr1.concat(arr2);
+const result = combined.includes(3);
+console.log(result);
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+false
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:308]:
```suggestion
`false`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 20]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+[1, 2, 3, 4, 5]
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:312]:
```suggestion
`[1, 2, 3, 4, 5]`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 21]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+undefined
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:316]:
```suggestion
`undefined`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 22]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+true
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:320]:
```suggestion
`true`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 23]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the output when using splice() to remove elements from a 2D array?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:354]:
```suggestion
What is the output when using `splice()` to remove elements from a 2D array?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 24]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+It gives the total number of elements in a 2D array.
 
 #### --answer--
 
-Placeholder answer
+It counts the number of rows in a 2D array.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:402]:
```suggestion
It counts the number of top-level elements in a 2D array.
```

Arrays don't really have rows.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md [hunk 25]
```diff
@@ -17,439 +17,505 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`push()` modifies the original array, while `concat()` creates a new array.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How does the rest parameter (`...rest`) work in array destructuring in JavaScript?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md:452]:
```suggestion
How does the rest parameter `...rest` work in array destructuring in JavaScript?
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Some opportunities to address, but this is a good start!

**Comment by Ritesh2351235** [comment]
@naomi-lgbt I have added the above-mentioned changes

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Pull request merged**